CONTRIBUTING.md

```diff
@@ -1,73 +1,130 @@
 
 1. Fork the repository.
-2. Modify the source; please focus on the specific change you are contributing.
-  If you also reformat all the code, it will be hard for us to focus on your change.
-3. Ensure local tests pass. (`colcon test` and `pre-commit run` (requires you to install pre-commit by `pip3 install pre-commit`)
+2. Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
+3. Ensure local tests pass. (``colcon test`` and ``pre-commit run`` (requires you to install pre-commit by ``pip3 install pre-commit``)
 4. Commit to your fork using clear commit messages.
 5. Send a pull request, answering any default questions in the pull request interface.
 6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
 
 GitHub provides additional documentation on [forking a repository](https://help.github.com/articles/fork-a-repo/) and
 [creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
 
+## Rules for the Repositories and Process of Merging Pull Requests
 
-## Finding contributions to work on
-Looking at the existing issues is a great way to find something to contribute on.
-As this project, by default, uses the default GitHub issue labels
-  (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any ['help wanted'][help-wanted] issues
-  is a great place to start.
+This section targets maintainers, but you are also welcome to read it to understand the process of how we handle PRs in our organization.
+This guideline is especially applicable for the following repositories:
+
+* ros2_control,
+* ros2_controllers,
+* ros2_control_demos.
+
+Please keep the following in mind:
+
+1. Please work from your fork when submitting PR. That way, we are keeping the main repo clean from feature branches.
+
+1. Each PR should have all checks satisfied before they can be considered for merging.
+
+1. Each PR must be approved by two maintainers (explicitly, please!). Only exceptions are PR's from other active maintainers in the repository, where one approval backed up with traceable discussion is sufficient.
+
+1. There is no need to do "squash and merge" of commits to your PR. We will squash the commits when merging the PR into the head branch.
+
+1. Please do not do "cowboy-style" PR merges over the weekend. It doesn't matter how trivial PR is. Give people a chance to do a proper review and comment on it.
+
+1. Be aware of the impact a PR has and give other maintainers and contributors sufficient time for the review proportional to its impact. Ping them if necessary, repeatedly if necessary.
+
+> Even if you are not a maintainer, you are still encouraged to review pull requests. This helps us increase the review pace and increase code quality. Also, you are very likely to find some issues/limitations nobody else is seeing.
```

Comment:
We add this to `..note`

Reply:
this is a markdown file, there is no ..note
This is why I used a blockquote here

Reply:
Got it